pkg/controller/nodeipam/ipam/multi_cidr_priority_queue_test.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipam
+
+import (
+	"container/heap"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/controller/nodeipam/ipam/multicidrset"
+	utilnet "k8s.io/utils/net"
+)
+
+func createTestPriorityQueueItem(name, cidr, selectorString string, labelMatchCount, perNodeHostBits int) *PriorityQueueItem {
+	_, clusterCIDR, _ := utilnet.ParseCIDRSloppy(cidr)
+	cidrSet, _ := multicidrset.NewMultiCIDRSet(clusterCIDR, perNodeHostBits)
+
+	return &PriorityQueueItem{
+		clusterCIDR: &multicidrset.ClusterCIDR{
+			Name:        name,
+			IPv4CIDRSet: cidrSet,
+		},
+		labelMatchCount: labelMatchCount,
+		selectorString:  selectorString,
+	}
+}
+
+func TestPriorityQueue(t *testing.T) {
+
+	pqi1 := createTestPriorityQueueItem("cidr1", "192.168.0.0/16", "foo=bar,name=test1", 1, 8)
+	pqi2 := createTestPriorityQueueItem("cidr2", "10.1.0.0/24", "foo=bar,name=test2", 2, 8)
+	pqi3 := createTestPriorityQueueItem("cidr3", "172.16.0.0/16", "foo=bar,name=test3", 2, 8)
+	pqi4 := createTestPriorityQueueItem("cidr4", "10.1.1.0/26", "abc=bar,name=test4", 2, 6)
+	pqi5 := createTestPriorityQueueItem("cidr5", "10.1.2.0/26", "foo=bar,name=test5", 2, 6)
+	pqi6 := createTestPriorityQueueItem("cidr6", "10.1.3.0/26", "abc=bar,name=test4", 2, 6)
+
+	for _, testQueue := range []struct {
+		name  string
+		items []*PriorityQueueItem
+		want  *PriorityQueueItem
+	}{
+		{"Test queue with single item", []*PriorityQueueItem{pqi1}, pqi1},
+		{"Test queue with items having different labelMatchCount", []*PriorityQueueItem{pqi1, pqi2}, pqi2},
+		{"Test queue with items having same labelMatchCount, different max Allocatable Pod CIDRs", []*PriorityQueueItem{pqi1, pqi2, pqi3}, pqi2},
+		{"Test queue with items having same labelMatchCount, max Allocatable Pod CIDRs, different PerNodeMaskSize", []*PriorityQueueItem{pqi1, pqi2, pqi4}, pqi4},
+		{"Test queue with items having same labelMatchCount, max Allocatable Pod CIDRs, PerNodeMaskSize, different labels", []*PriorityQueueItem{pqi1, pqi2, pqi4, pqi5}, pqi4},
+		{"Test queue with items having same labelMatchCount, max Allocatable Pod CIDRs, PerNodeMaskSize, labels, different IP addresses", []*PriorityQueueItem{pqi1, pqi2, pqi4, pqi5, pqi6}, pqi4},
+	} {
```

Comment:
if you use ` t.Run(testname, func(t *testing.T) {` you get better isolation https://go.dev/blog/subtests , just FYI, not need to change it

Reply:
+1

Reply:
Done